python/foglamp/services/core/api/common.py

```diff
@@ -99,9 +99,10 @@ def get_stats(k):
         raise web.HTTPException(reason=str(ex))
 
 
-def do_shutdown(loop=None):
+def do_shutdown(request):
     _logger.info("Executing controlled shutdown")
-    if loop is None:
-        loop = asyncio.get_event_loop()
-    loop.run_until_complete(server.Server._stop())
-    loop.stop()
+    try:
+        loop = request.loop
+        loop.run_until_complete(server.Server.shutdown(request))
+    except RuntimeError as ex:
+        _logger.exception("Error while stopping foglamp server: {}".format(str(ex)))
```

Comment:
foglamp -> FogLAMP
RuntimeError as `ex` > RuntimeError as `e`

Reply:
I think we should also raise, not only just log! _stop has bare exception raise for the same purpose